### PR TITLE
[Merged by Bors] - feat: in a unital C⋆-algebra, `‖a‖ ≤ r ↔ a ≤ r • 1`, and related lemmas

### PR DESCRIPTION
**Pull request by j-loreaux**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/Unital.lean [hunk 1]
```diff
@@ -898,6 +898,14 @@ lemma one_le_cfc (f : R → R) (a : A) (h : ∀ x ∈ spectrum R a, 1 ≤ f x)
     1 ≤ cfc f a := by
   simpa using algebraMap_le_cfc f 1 a h
 
+lemma CFC.one_le {a : A} (h : ∀ x ∈ spectrum R a, x ≤ 1) (ha : p a := by cfc_tac) :
+    a ≤ 1 := by
+  simpa using le_algebraMap_of_spectrum_le h
+
+lemma CFC.le_one {a : A} (h : ∀ x ∈ spectrum R a, 1 ≤ x) (ha : p a := by cfc_tac) :
+    1 ≤ a := by
```
Comment (dupuisf) [inline Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/Unital.lean:906]: Shouldn't we swap the names of those two?

File: Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/Unital.lean [hunk 2]
```diff
@@ -939,6 +947,44 @@ lemma cfc_nonpos_iff (f : R → R) (a : A) (hf : ContinuousOn f (spectrum R a) :
...
+    (hf : ContinuousOn f (spectrum R a) := by cfc_cont_tac) (ha : p a := by cfc_tac) :
+    1 ≤ cfc f a ↔ ∀ x ∈ spectrum R a, 1 ≤ f x := by
+  simpa using algebraMap_le_cfc_iff f 1 a
+
+lemma CFC.one_le_iff (a : A) (ha : p a := by cfc_tac) :
+    a ≤ 1 ↔ ∀ x ∈ spectrum R a, x ≤ 1 := by
+  simpa using le_algebraMap_iff_spectrum_le (r := (1 : R)) (a := a)
+
+lemma CFC.le_one_iff (a : A) (ha : p a := by cfc_tac) :
+    1 ≤ a ↔ ∀ x ∈ spectrum R a, 1 ≤ x := by
```
Comment (dupuisf) [inline Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/Unital.lean:985]: Again, the names should be swapped

File: Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/Order.lean [hunk 3]
```diff
@@ -103,24 +103,58 @@ lemma IsSelfAdjoint.toReal_spectralRadius_eq_norm {a : A} (ha : IsSelfAdjoint a)
...
+    ‖a‖ ≤ 1 ↔ a ≤ 1 := by
+  simpa using norm_le_iff_of_nonneg a zero_le_one
+
+lemma nnnorm_le_one_iff_of_nonneg (a : A) (ha : 0 ≤ a := by cfc_tac) :
+    ‖a‖₊ ≤ 1 ↔ a ≤ 1 := by
+  rw [← NNReal.coe_le_coe]
+  exact norm_le_one_iff_of_nonneg a
+
+lemma norm_le_ofNat_iff_of_nonneg (a : A) (n : ℕ) (ha : 0 ≤ a := by cfc_tac) :
+    ‖a‖ ≤ n ↔ a ≤ n := by
```
[the remark this answers is not in the document]
Comment (j-loreaux) [reply Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/Order.lean]: I don't think we need the `ofNat` version, I just messed up the naming when I added these.

**Comment by j-loreaux** [comment]
@dupuisf just so you're aware, I not only fixed the lemma names, but I also added a few new lemmas too.

File: Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/Order.lean [hunk 4]
```diff
@@ -64,6 +64,44 @@ lemma inr_nonneg_iff {a : A} : 0 ≤ (a : Unitization ℂ A) ↔ 0 ≤ a := by
 
 end Unitization
 
+lemma ContinuousOn.ofReal_map_toNNReal {f : ℝ≥0 → ℝ≥0} {s : Set ℝ} {t : Set ℝ≥0}
+    (hf : ContinuousOn f t) (h : Set.MapsTo Real.toNNReal s t) :
```
Comment (dupuisf) [inline Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/Order.lean:68]: `#find_home` says `Topology.Instances.NNReal` which seems pretty reasonable.

**Review by dupuisf — COMMENTED** [review]
Just a few last comments, but otherwise LGTM.

bors d+

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**